### PR TITLE
etcdserver: Move put/read/delete on Alarm bucket to bucket package

### PR DESCRIPTION
**Pull request by serathius**
cc @ptabor

### CODE REVIEW

File: server/etcdserver/api/v3alarm/alarms.go [hunk 1]
```diff
@@ -59,16 +59,10 @@ func (a *AlarmStore) Activate(id types.ID, at pb.AlarmType) *pb.AlarmMember {
...
-		a.lg.Panic("failed to marshal alarm member", zap.Error(err))
-	}
-
-	b := a.bg.Backend()
-	b.BatchTx().Lock()
-	b.BatchTx().UnsafePut(buckets.Alarm, v, nil)
-	b.BatchTx().Unlock()
-
+	tx := a.bg.Backend().BatchTx()
+	tx.Lock()
```
Comment (ptabor) [inline server/etcdserver/api/v3alarm/alarms.go:63]: Seems it should be  (safe) `buckets.MustPutAlarm(a.lg, tx, newAlarm)`

File: server/etcdserver/api/v3alarm/alarms.go [hunk 2]
```diff
@@ -88,16 +82,10 @@ func (a *AlarmStore) Deactivate(id types.ID, at pb.AlarmType) *pb.AlarmMember {
...
-
-	b := a.bg.Backend()
-	b.BatchTx().Lock()
-	b.BatchTx().UnsafeDelete(buckets.Alarm, v)
-	b.BatchTx().Unlock()
-
+	tx := a.bg.Backend().BatchTx()
+	tx.Lock()
+	defer tx.Unlock()
+	buckets.MustUnsafeDeleteAlarm(a.lg, tx, m)
```
Comment (ptabor) [inline server/etcdserver/api/v3alarm/alarms.go:88]: Please put it together as `buckets.MustDeleteAlarm(...)`

File: server/etcdserver/api/v3alarm/alarms.go [hunk 3]
```diff
@@ -123,15 +111,14 @@ func (a *AlarmStore) restore() error {
...
-		if err := m.Unmarshal(k); err != nil {
-			return err
-		}
-		a.addToMap(&m)
-		return nil
-	})
+	buckets.UnsafeCreateAlarmBucket(tx)
+	ms, err := buckets.UnsafeGetAllAlarms(tx)
+	if err != nil {
+		return err
```
Comment (ptabor) [inline server/etcdserver/api/v3alarm/alarms.go:117]: Are we missing here tx.Unlock() ? 
probably good reason to move it to a separate method with a 'defer' block

**Review by ptabor — CHANGES_REQUESTED** [review]

File: server/mvcc/buckets/alarm.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
+// Copyright 2016 The etcd Authors
```
Comment (serathius) [inline server/mvcc/buckets/alarm.go:1]: Heh you didn't noticed this time :P

File: server/etcdserver/api/v3alarm/alarms.go (hunk 3 above)
Comment (serathius) [reply server/etcdserver/api/v3alarm/alarms.go:111]: Done.

File: server/etcdserver/api/v3alarm/alarms.go (hunk 2 above)
Comment (serathius) [reply server/etcdserver/api/v3alarm/alarms.go]: Done

File: server/etcdserver/api/v3alarm/alarms.go (hunk 1 above)
Comment (serathius) [reply server/etcdserver/api/v3alarm/alarms.go]: Done

**Pull request merged**